File: src/test/java/SquareGraphTest.java
import org.junit.Test;

import static org.junit.Assert.assertArrayEquals;
import static org.junit.Assert.assertEquals;

public class SquareGraphTest {

    @Test
    public void createGraph() {
        new SquareGraph(15);
    }

    @Test
    public void knowsEdgesForNode() {
        SquareGraph graph = new SquareGraph(15);

        assertArrayEquals(new int[]{15, 8, 3}, graph.getNeighbors(1));
        assertArrayEquals(new int[]{9, 2}, graph.getNeighbors(7));
        assertArrayEquals(new int[]{7}, graph.getNeighbors(9));
        assertArrayEquals(new int[]{10, 1}, graph.getNeighbors(15));
    }

    @Test
    public void knowsDegreeOfVertex() {
        SquareGraph graph = new SquareGraph(15);

        assertEquals(3, graph.degree(1));
        assertEquals(1, graph.degree(9));
        assertEquals(2, graph.degree(15));
    }
}

File: src/test/java/search/SquareGraphTest.kt
package search

import org.junit.Assert
import org.junit.Test

class SquareGraphTest {

    @Test
    fun `creates Graph`() {
        SquareGraph(15)
    }

    @Test
    fun `knows Edges For Node`() {
        val graph = SquareGraph(15)

        Assert.assertArrayEquals(intArrayOf(15, 8, 3), graph.getNeighbors(1))
        Assert.assertArrayEquals(intArrayOf(9, 2), graph.getNeighbors(7))
        Assert.assertArrayEquals(intArrayOf(7), graph.getNeighbors(9))
        Assert.assertArrayEquals(intArrayOf(10, 1), graph.getNeighbors(15))
    }

    @Test
    fun `knows Degree Of Vertex`() {
        val graph = SquareGraph(15)

        Assert.assertEquals(3, graph.degree(1).toLong())
        Assert.assertEquals(1, graph.degree(9).toLong())
        Assert.assertEquals(2, graph.degree(15).toLong())
    }
}
